src/validators/DateInputValidator.js

```diff
@@ -0,0 +1,103 @@
+import dayjs from '../helpers/getDayjs.js'
+
+export const DATE_FORMATS = ['mm/dd/yyyy', 'mm/yyyy', 'mm/yy']
+
+export const dateMaskByFormat = {
+  'mm/dd/yyyy': [/\d/, /\d/, '/', /\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
+  'mm/yyyy': [/\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
+  'mm/yy': [/\d/, /\d/, '/', /\d/, /\d/],
+}
+
+export const dateRegexByFormat = {
+  'mm/dd/yyyy': /\d\d\/\d\d\/\d\d\d\d/,
+  'mm/yyyy': /\d\d\/\d\d\d\d/,
+  'mm/yy': /\d\d\/\d\d/,
```

Comment:
nit: This could be cleaned up with the quantifier `/\d{1,4}/` capturing 1-4 digits for example.  If capture groups are included it could also be used for extracting data/formatting the input (if you make the `\` optional be rewrite it... something like `/(?<month>\d{1,2})(?:\/?)(?<year>\d{0,4})/`, which would give named capture groups too, and slash being a special non-capturing group.  The other part of it is that this regex could be tokenized into its components, then the true regex could be composed, eg:
```js
  const month: `(?<month>\d{0,2})`
  const year: `(?<year>\d{0,4})`
  const slash: `(?:\/?)`

  const mmyyyyRegex = new Regex(month + slash + year)
```

Reply:
Brian, I really really like you're point here and I totally think it has merit. Let me tell you why I'd like to punt on this though.

This refactor already feels a bit aggressive as I'm completely teasing apart all the BirthdateInput code and delegating. I've captured your sentiments though:
https://app.asana.com/0/1137984670709817/1176569816350953/f

Reply:
That's completely sufficient, as it would definitely be an optimization